SimCalorimetry/HGCalSimProducers/plugins/alpaka/HGCalDigiSoaFiller.cc

```diff
@@ -0,0 +1,185 @@
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDPutToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESGetToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/Event.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/CopyToDevice.h"
+
+#include "DataFormats/HGCalDigi/interface/HGCalElectronicsId.h"
+#include "DataFormats/HGCDigi/interface/HGCDigiCollections.h"
+#include "DataFormats/HGCalDigi/interface/HGCalDigiHost.h"
+#include "DataFormats/HGCalDigi/interface/alpaka/HGCalDigiDevice.h"
+#include "CondFormats/DataRecord/interface/HGCalElectronicsMappingRcd.h"
+#include "CondFormats/HGCalObjects/interface/HGCalMappingModuleIndexer.h"
+//#include "CondFormats/HGCalObjects/interface/HGCalMappingCellIndexer.h"
+#include "CondFormats/HGCalObjects/interface/alpaka/HGCalMappingParameterDevice.h"
+#include "Geometry/HGCalMapping/interface/HGCalMappingTools.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+
+  /**
+   *   @short consumes channel digis and packs then in a realistic FEDRawData stream
+  */
+  class HGCalDigiSoaFiller : public stream::EDProducer<> {
+  public:
+    //using CellIndexer = HGCalMappingCellIndexer;
+    using CellInfo = hgcal::HGCalMappingCellParamDevice;
+    using ModuleIndexer = HGCalMappingModuleIndexer;
+    using ModuleInfo = hgcal::HGCalMappingModuleParamDevice;
+
+    /**
+      @  short constructor
+    */
+    explicit HGCalDigiSoaFiller(edm::ParameterSet const& config);
+
+    /**
+       @short parameters to be used with this plugin
+    */
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+    /**
+      @short steers the conversion to SOA
+    */
+    void produce(device::Event&, device::EventSetup const&) override;
+
+  private:
+    edm::EDGetTokenT<edm::SortedCollection<HGCalDataFrame> > digisCEET_, digisCEHSiT_, digisCEHSiPMT_;
+    device::EDPutToken<hgcaldigi::HGCalDigiDevice> digiProdT_;
+
+    edm::ESWatcher<HGCalElectronicsMappingRcd> cfgWatcher_;
+    //edm::ESGetToken<CellIndexer, HGCalElectronicsMappingRcd> cellIndexTkn_;
+    device::ESGetToken<CellInfo, HGCalElectronicsMappingRcd> cellTkn_;
+    edm::ESGetToken<ModuleIndexer, HGCalElectronicsMappingRcd> moduleIndexTkn_;
+    device::ESGetToken<ModuleInfo, HGCalElectronicsMappingRcd> moduleTkn_;
+  };
+
+  //
+  HGCalDigiSoaFiller::HGCalDigiSoaFiller(const edm::ParameterSet& config)
+      : digisCEET_(
+            consumes<edm::SortedCollection<HGCalDataFrame> >(config.getParameter<edm::InputTag>("eeChannelDigis"))),
+        digisCEHSiT_(
+            consumes<edm::SortedCollection<HGCalDataFrame> >(config.getParameter<edm::InputTag>("cehsiChannelDigis"))),
+        digisCEHSiPMT_(consumes<edm::SortedCollection<HGCalDataFrame> >(
+            config.getParameter<edm::InputTag>("cehsipmChannelDigis"))),
```

Comment:
These could be
```suggestion
      : digisCEET_(
            consumes(config.getParameter<edm::InputTag>("eeChannelDigis"))),
        digisCEHSiT_(
            consumes(config.getParameter<edm::InputTag>("cehsiChannelDigis"))),
        digisCEHSiPMT_(consumes(config.getParameter<edm::InputTag>("cehsipmChannelDigis"))),
```

Reply:
This file was remove all together in https://github.com/CMS-HGCAL/cmssw/pull/137.